src/js/components/search/filters/timePeriod/TimePeriod.jsx

```diff
@@ -143,23 +148,34 @@ export default class TimePeriod extends React.Component {
             activeClassDR = '';
         }
 
-        return (
-            <div className="time-period-filter">
-                <b>Time Period</b>
+        let timePeriod = null;
+        if (this.props.showFilter === true) {
+            timePeriod = (<div>
```

Comment:
This is too much JSX in the logic; refactor this code so the bulk of the JSX is in the return rather than in a variable. This may involve rethinking how you've restructured your components.